File: AssignmentService/frontend/src/app/model/external-response.ts
export class ExternalResponse {
  constructor(mediaTitle: string, mediaCreators: string[], mediaType: string, mediaSource: string, artWork: string) {
    this._mediaTitle = mediaTitle;
    this._mediaCreators = mediaCreators;
    this._mediaType = mediaType;
    this._mediaSource = mediaSource;
    this._artWork = artWork;
  }

  private _mediaTitle: string;

  get mediaTitle(): string {
    return this._mediaTitle;
  }

  set mediaTitle(value: string) {
    this._mediaTitle = value;
  }

  private _mediaCreators: string[];

  get mediaCreators(): string[] {
    return this._mediaCreators;
  }

  set mediaCreators(value: string[]) {
    this._mediaCreators = value;
  }

  private _mediaType: string;

  get mediaType(): string {
    return this._mediaType;
  }

  set mediaType(value: string) {
    this._mediaType = value;
  }

  private _mediaSource: string;

  get mediaSource(): string {
    return this._mediaSource;
  }

  set mediaSource(value: string) {
    this._mediaSource = value;
  }

  private _artWork: string;

  get artWork(): string {
    return this._artWork;
  }

  set artWork(value: string) {
    this._artWork = value;
  }
}

File: AssignmentService/src/main/java/com/idexx/assignmentservice/model/ExternalResponse.java
package com.idexx.assignmentservice.model;

import java.util.List;

public class ExternalResponse {
    private String mediaTitle;
    private List<String> mediaCreators;
    private String mediaType;
    private String mediaSource;
    private String artWork;

    public ExternalResponse() {
    }


    public ExternalResponse(String mediaTitle, List<String> mediaCreators, String mediaType, String mediaSource, String artWork) {
        this.mediaTitle = mediaTitle;
        this.mediaCreators = mediaCreators;
        this.mediaType = mediaType;
        this.mediaSource = mediaSource;
        this.artWork = artWork;
    }

    public String getArtWork() {
        return artWork;
    }

    public void setArtWork(String artWork) {
        this.artWork = artWork;
    }

    public String getMediaTitle() {
        return mediaTitle;
    }

    public void setMediaTitle(String mediaTitle) {
        this.mediaTitle = mediaTitle;
    }

    public List<String> getMediaCreators() {
        return mediaCreators;
    }

    public void setMediaCreators(List<String> mediaCreators) {
        this.mediaCreators = mediaCreators;
    }

    public String getMediaType() {
        return mediaType;
    }

    public void setMediaType(String mediaType) {
        this.mediaType = mediaType;
    }

    public String getMediaSource() {
        return mediaSource;
    }

    public void setMediaSource(String mediaSource) {
        this.mediaSource = mediaSource;
    }
}
